addons/mail/static/src/components/notification_list/notification_list.js

```diff
@@ -133,9 +133,8 @@ class NotificationList extends Component {
         if (props.filter === 'all') {
             const notificationGroups = this.env.messaging.notificationGroupManager.groups;
             notifications = Object.values(notificationGroups)
-                .sort((group1, group2) =>
-                    group1.date.isAfter(group2.date) ? -1 : 1
-                ).map(notificationGroup => {
+                .sort((group1, group2) => group1.sequence > group2.sequence ? 1 : -1)
```

Comment:
Doesn't really matter and it was already wrong before, but ideally this should return 0 if both groups have the same sequence.